compiler/util/misc.cpp

```diff
@@ -160,10 +160,10 @@ bool requireOutlinedOn() {
 }
 
 const char* cleanFilename(const char* name) {
-  static int  chplHomeLen = strlen(CHPL_HOME);
+  static int  chplHomeLen = CHPL_HOME.length();
   const char* retval      = NULL;
 
-  if (strncmp(name, CHPL_HOME, chplHomeLen) == 0) {
+  if (strncmp(name, CHPL_HOME.c_str(), chplHomeLen) == 0) {
     retval = astr("$CHPL_HOME", name + chplHomeLen);
```

Comment:
I think this could be made a little cleaner in general by using `string_view(name)`, and then using more modern C++ to check the prefix of the string, rather than relying on `strncmp`. But this is more of nit, what is here looks fine to me

Reply:
Went ahead and did it, I've never used `string_view` before and now's a fine time to try it out. I also went ahead and used it for the driver save/restore functions, which we sometimes want to use with cstrings and sometimes with `std::strings`; it looks marginally cleaner to me now.